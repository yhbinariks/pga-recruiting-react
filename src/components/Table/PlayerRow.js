import React from 'react';
import PropTypes from 'prop-types';

const PlayerRow = ({ player, onEdit, onDelete }) => (
  <tr>
    <td>{`${player.lastName}, ${player.firstName}`}</td>
    <td>{player.score}</td>
    <td>
      <button
        type="button"
        className="table__action-btn"
        onClick={onEdit('edit', player.id)}
      >
        Edit
      </button>
      <button
        type="button"
        className="table__action-btn"
        onClick={onDelete(player.id)}
      >
        Delete
      </button>
    </td>
  </tr>
);

const propTypes = {
  onEdit: PropTypes.func.isRequired,
  onDelete: PropTypes.func.isRequired,
  player: PropTypes.shape({
    lastName: PropTypes.string.isRequired,
    firstName: PropTypes.string.isRequired,
    score: PropTypes.number.isRequired,
  }).isRequired,
};

PlayerRow.propTypes = propTypes;

export default PlayerRow;
