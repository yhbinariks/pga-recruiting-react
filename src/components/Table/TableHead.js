import React from 'react';
import PropTypes from 'prop-types';

const TableHead = ({ setSortBy }) => (
  <thead>
    <tr>
      <th onClick={() => setSortBy('name')}>Name</th>
      <th onClick={() => setSortBy('score')}>Score</th>
      <th />
    </tr>
  </thead>
);

const propTypes = {
  setSortBy: PropTypes.func.isRequired,
};

TableHead.propTypes = propTypes;

export default TableHead;
