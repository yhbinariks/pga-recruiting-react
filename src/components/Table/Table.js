import React from 'react';
import PropTypes from 'prop-types';
import PlayerRow from './PlayerRow';
import TableHead from './TableHead';

const Table = ({
  players,
  onDelete,
  setSortBy,
  changeMode,
}) => (
  <table>
    <TableHead setSortBy={setSortBy} />
    <tbody>
      {players.map(player => (
        <PlayerRow
          key={player.id}
          player={player}
          onDelete={onDelete}
          onEdit={changeMode}
        />
      ))}
    </tbody>
  </table>
);

const propTypes = {
  players: PropTypes.arrayOf(PropTypes.shape({
    lastName: PropTypes.string.isRequired,
    firstName: PropTypes.string.isRequired,
    score: PropTypes.number.isRequired,
    id: PropTypes.string.isRequired,
  })).isRequired,
  changeMode: PropTypes.func.isRequired,
  setSortBy: PropTypes.func.isRequired,
  onDelete: PropTypes.func.isRequired,
};

Table.propTypes = propTypes;

export default Table;
