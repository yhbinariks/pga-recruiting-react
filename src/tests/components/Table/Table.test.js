import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';
import Table from '../../../components/Table/Table';

describe('Table component', () => {
  it('snapshot', () => {
    const props = {
      players: [
        {
          firstName: 'petro',
          lastName: 'chu',
          score: 32,
          id: '123',
        },
        {
          firstName: 'petro2',
          lastName: 'chu',
          score: 32,
          id: '1231',
        },
        {
          firstName: 'petro3',
          lastName: 'chu3',
          score: 32,
          id: '1233',
        },
      ],
      changeMode: jest.fn(),
      setSortBy: jest.fn(),
      onDelete: jest.fn(),
    };
    const table = shallow(<Table {...props} />);
    expect(toJson(table)).toMatchSnapshot();
  });
});
