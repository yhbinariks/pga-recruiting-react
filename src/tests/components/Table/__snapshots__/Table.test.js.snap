// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table component snapshot 1`] = `
<table>
  <TableHead
    setSortBy={[MockFunction]}
  />
  <tbody>
    <PlayerRow
      key="123"
      onDelete={[MockFunction]}
      onEdit={[MockFunction]}
      player={
        Object {
          "firstName": "petro",
          "id": "123",
          "lastName": "chu",
          "score": 32,
        }
      }
    />
    <PlayerRow
      key="1231"
      onDelete={[MockFunction]}
      onEdit={[MockFunction]}
      player={
        Object {
          "firstName": "petro2",
          "id": "1231",
          "lastName": "chu",
          "score": 32,
        }
      }
    />
    <PlayerRow
      key="1233"
      onDelete={[MockFunction]}
      onEdit={[MockFunction]}
      player={
        Object {
          "firstName": "petro3",
          "id": "1233",
          "lastName": "chu3",
          "score": 32,
        }
      }
    />
  </tbody>
</table>
`;
